File: day1/day1++/puzzles.cpp
#include <algorithm>
#include <fstream>
#include <functional>
#include <iostream>
#include <stdio.h>
#include <string>
#include <vector>

using std::vector;

int main() {
  vector<int> elfCals;
  std::ifstream file;
  file.open("../elfCals.txt");

  int sum{0};
  std::string line;
  while (std::getline(file, line)) {
    if (line.empty()) {
      elfCals.push_back(sum);
      sum = 0;
    } else {
      sum += std::stoi(line);
    }
  }

  std::sort(elfCals.begin(), elfCals.end(), std::greater<int>());
  printf("Answer1: %i\n", elfCals[0]);
  printf("Answer2: %i\n", elfCals[0] + elfCals[1] + elfCals[2]);

  file.close();
}

File: day1/Puzzles.cs
List<int> elfCals = new();
int currCals = 0;

foreach (var line in File.ReadAllLines("elfCals.txt")) {
    if (int.TryParse(line, out int value)) {
        currCals += value;
    } else {
        elfCals.Add(currCals);
        currCals = 0;
    }
}

elfCals.Sort((a, b) => b.CompareTo(a));
int total = elfCals[0] + elfCals[1] + elfCals[2];
Console.WriteLine($"Answer1: {elfCals[0]}");
Console.WriteLine($"Answer2: {total}");
